From 03212d87b4b8ff5a09c80106d1e21e15a9869586 Mon Sep 17 00:00:00 2001
From: Anthony Loiseau <thannoy@actech-innovation.com>
Date: Tue, 28 Jul 2009 16:46:54 +0200
Subject: [PATCH 1/3] codec/libass: avoid possible segfault
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

dialog_ProgressCreate can return NULL. For this case,
only deal with the result if it is not NULL.

Signed-off-by: Rémi Duraffort <ivoire@videolan.org>
(cherry picked from commit 29e7fc71082c08aec51d438b3d8564e25526b2d1)

Signed-off-by: Rémi Duraffort <ivoire@videolan.org>
---
 modules/codec/libass.c |   12 ++++++++----
 1 files changed, 8 insertions(+), 4 deletions(-)

Index: vlc-1.0.1/modules/codec/libass.c
===================================================================
--- vlc-1.0.1.orig/modules/codec/libass.c
+++ vlc-1.0.1/modules/codec/libass.c
@@ -731,7 +731,8 @@ static ass_handle_t *AssHandleHold( deco
     ass_set_fonts_dir( p_library, psz_font_dir );
     free( psz_font_dir );
 #ifdef WIN32
-    dialog_ProgressSet( p_dialog, NULL, 0.1 );
+    if( p_dialog )
+        dialog_ProgressSet( p_dialog, NULL, 0.1 );
 #endif
 
     ass_set_extract_fonts( p_library, true );
@@ -754,7 +755,8 @@ static ass_handle_t *AssHandleHold( deco
 
 #ifdef HAVE_FONTCONFIG
 #ifdef WIN32
-    dialog_ProgressSet( p_dialog, NULL, 0.2 );
+    if( p_dialog )
+        dialog_ProgressSet( p_dialog, NULL, 0.2 );
 #endif
 #if defined( LIBASS_VERSION ) && LIBASS_VERSION >= 0x00907000
     ass_set_fonts( p_renderer, psz_font, psz_family, true, NULL, 1 );  // setup default font/family
@@ -762,7 +764,8 @@ static ass_handle_t *AssHandleHold( deco
     ass_set_fonts( p_renderer, psz_font, psz_family );  // setup default font/family
 #endif
 #ifdef WIN32
-    dialog_ProgressSet( p_dialog, NULL, 1.0 );
+    if( p_dialog )
+        dialog_ProgressSet( p_dialog, NULL, 1.0 );
 #endif
 #else
     /* FIXME you HAVE to give him a font if no fontconfig */
@@ -781,7 +784,8 @@ static ass_handle_t *AssHandleHold( deco
     /* */
     vlc_mutex_unlock( &libass_lock );
 #ifdef WIN32
-    dialog_ProgressDestroy( p_dialog );
+    if( p_dialog )
+        dialog_ProgressDestroy( p_dialog );
 #endif
     return p_ass;
 
